### PR TITLE
[RFC] Feedback problem

### PR DESCRIPTION
**Pull request by gfokkema**

### CODE REVIEW

File: final/problem-def-ana/problem-ana.tex [hunk 1]
```diff
@@ -1,8 +1,7 @@
 \section{Problem Analysis}
 \label{sec:problem-analysis}
-This section gives an analysis of the problem defined in the previous
-section. More specifically, this section describes the problems that
-need to be solved while incorporating a REPL within the larger context of Spoofax.
+This section section describes the problems that need to be solved while
```
Comment (Hjdskes) [inline final/problem-def-ana/problem-ana.tex]: sub-problems? To stay in theme with the rest of this section.

Comment (Hjdskes) [inline final/problem-def-ana/problem-ana.tex]: Double `section`

File: final/problem-def-ana/problem-ana.tex [hunk 2]
```diff
@@ -14,42 +13,33 @@ \section{Problem Analysis}
...
-language. For example, some languages have an analysis step, whereas
+Different Spoofax languages have different ways of executing a program.
+For example, some languages have an analysis step, whereas
 others do not. Moreover, specifying the dynamic semantics of a language
 can be done in multiple ways (see \cref{ssec:dynamic-semantics}): one
 can use DynSem, Stratego, and some languages even implement a Java
 backend\footnote{See IceDust:
   \url{https://github.com/MetaBorgCube/IceDust}}.
 
 Some of these properties of a language should be detected by the REPL
```
Comment (Hjdskes) [inline final/problem-def-ana/problem-ana.tex]: This shows a requirement and therefore has to go:

> Some of these properties of a language should be detected by the REPL automatically, such as whether a language has an analysis step.

Comment (Hjdskes) [inline final/problem-def-ana/problem-ana.tex]: The remainder of this paragraph has the same problem.

File: final/problem-def-ana/problem-ana.tex [hunk 3]
```diff
@@ -14,42 +13,33 @@ \section{Problem Analysis}
...
 precisely what an execution environment looks like differs among
-languages. This is therefore an area where the language designer
-should configure the REPL so that it does the evaluation the way they
-want. To accommodate for this, the REPL should provide a configuration
-interface that is generic enough for the REPL to be configured
-for any language.
+languages.
 
-\subsection{Language-specific additional commands}
+\subsection{Language-specific commands}
```
Comment (Hjdskes) [inline final/problem-def-ana/problem-ana.tex:38]: I think having `additional` here emphasizes that they are defined outside of the core backend.

File: final/problem-def-ana/problem-ana.tex [hunk 4]
```diff
@@ -81,21 +63,10 @@ \subsection{REPL-specific semantics}
...
-to define a different semantics in the context of a REPL session. For
-example, in the case of a language that uses DynSem for specifying its
-dynamic semantics, the language designer can provide additional rules
-that are meant only for the REPL. Again, the REPL should provide a
-flexible configuration interface for these REPL-specific
-semantics. Another desirable property of this configuration interface
-is to allow the language designer to reuse the semantics that are
-common both in the context of a REPL session and outside of it.
+This poses a similar problem as in the previous section: adding REPL-specific
+semantics to the language should not affect the original semantics of the
```
Comment (Hjdskes) [inline final/problem-def-ana/problem-ana.tex:63]: Again: `should` implies requirement; does not belong here.

File: final/design/overview.tex [hunk 5]
```diff
@@ -3,26 +3,29 @@ \section{Overview}
...
-
-The \texttt{ICommandInvoker} instance maintains a map from command names to
-corresponding \texttt{IReplCommand} instances. A frontend can thus send
-input directly to the \texttt{execute} method, after which the command
-corresponding to the input is resolved and executed. The executed command
-then returns an instance of a class implementing the \texttt{IResult}
-interface. The actual type of the returned class depends on the executed command
-and whether the command was successful or not. The frontend is itself
-responsible for deciding how to present the result types to the user.
+its design can be found in \cref{fig:uml-overview}. A diagram of the complete
```
Comment (Hjdskes) [inline final/design/overview.tex:8]: `s/A diagram/The UML/`

File: final/design/overview.tex [hunk 6]
```diff
@@ -3,26 +3,29 @@ \section{Overview}
...
+by sending the command name and its parameters to the \texttt{ICommandInvoker}
+in the backend. The \texttt{ICommandInvoker} will then execute the
+\texttt{IReplCommand} corresponding to the command name. Every executed command
+returns an \texttt{IResult}. Since execution of different commands can have
+varying types of results, the actual returned type of \texttt{IResult} can
+vary.
+
+To execute a command, the frontend thus only has to implement a way of querying
+the user for input, after which it can send this input to the
+\texttt{ICommandInvoker}. The returned \texttt{IResult} can then be visited
```
Comment (Hjdskes) [inline final/design/overview.tex]: I would move the "The returned ... specific types" sentence to the end of the previous paragraph.

File: final/problem-def-ana/problem-ana.tex (hunk 2 above)
Comment (gfokkema) [inline final/problem-def-ana/problem-ana.tex]: Do not necessarily agree here.

Comment (Hjdskes) [inline final/problem-def-ana/problem-ana.tex]: This isn't my idea but Hendrik's

File: final/design/overview.tex (hunk 5 above)
Comment (gfokkema) [inline final/design/overview.tex:8]: Actually it's a UML diagram, so UML in itself is invalid, as opposed to diagram, which is valid.

**Pull request merged**